crates/extensions/c8y_mapper_ext/src/config_operations.rs

```diff
@@ -214,8 +214,9 @@ impl CumulocityConverter {
         let download = match download_result {
             Err(err) => {
                 let smartrest_error = SmartRestSetOperationToFailed::new(
-                    CumulocitySupportedOperations::C8yLogFileRequest,
-                    format!("Upload failed with {:?}", err),
+                    // TODO test I'm the right message type
```

Comment:
I was surprised to see this was how we handle failed operations. We must already know this value is order to execute the operation, so all this function should need to provide is the failure reason, it shouldn't care what the operation is called. This is a particularly critical thing to handle correctly due to Cumulocity's assumption that devices always complete operations, and reproducing this failure is pretty difficult in order to test it (and expensive to test since it requires an integration test if you want more of a guarantee than "hopefully the test author wrote the correct value in their test").

Reply:
> We must already know this value is order to execute the operation, so all this function should need to provide is the failure reason, it shouldn't care what the operation is called.

It's partly due to the fact that the operation status transitions in SmartREST is [generic](https://cumulocity.com/guides/reference/smartrest-two/#501) and requires the operation type to be sent in that message. That same design was adopted here as well, in generic structures like `SmartRestSetOperationToFailed` which leads to such accidental errors. But, I agree that some sort of "current-operation-aware" warpper should have been used here to avoid such mistakes.